src/runner.ts

```diff
@@ -173,25 +185,80 @@ async function runLinter(options: Options, logger: Logger): Promise<LintResult>
     return doLinting(options, files, program, logger);
 }
 
-function resolveFilesAndProgram(
-    { files, project, exclude, outputAbsolutePaths }: Options,
+/** Read the stdin stream into a string with all of the contents. */
+async function readStdin(): Promise<string> {
+    return new Promise<string>((resolve, reject) => {
+        let fileData = "";
+
+        process.stdin.setEncoding("utf8");
+
+        process.stdin.on("readable", () => {
+            let chunk: string | Buffer | null = null;
+
+            // tslint:disable-next-line: no-conditional-assignment
+            while ((chunk = process.stdin.read()) !== null) {
+                fileData += chunk;
+            }
+        });
+
+        process.stdin.on("end", () => {
+            resolve(fileData);
+        });
+
+        process.stdin.on("error", (error: Error) => {
+            reject(error);
+        });
+    });
+}
+
+interface ResolvedFile {
+  filename: string;
+  isStdin: boolean;
+}
+
+function resolveProgram(
+    { project }: Options,
+): ts.Program | undefined {
+    if (project !== undefined) {
+        const projectPath = findTsconfig(project);
+
+        if (projectPath === undefined) {
+            throw new FatalError(`Invalid option for project: ${project}`);
+        }
+
+        return Linter.createProgram(projectPath);
+    }
+
+    return undefined;
+}
+
+function resolveFiles(
+    { files, stdin, stdinFilename, exclude, outputAbsolutePaths }: Options,
+    program: ts.Program | undefined,
     logger: Logger,
-): { files: string[]; program?: ts.Program } {
+): ResolvedFile[] {
+    // Return stdin files if we're reading from stdin.
+    if (stdin) {
+        if (stdinFilename) {
+            // Use a filename for shadowing stdin, if the option is set.
+            return [{filename: path.resolve(stdinFilename), isStdin: true}];
+        } else {
+            return [{filename: "", isStdin: true}];
```

Comment:
maybe for now you should fail hard here with a good error message, until `--stdin` w/o `--stdinFilename` is supported